src/garage/envs/environment.py

```diff
@@ -0,0 +1,188 @@
+"""Base Garage Environment API."""
+
+import abc
+
+
+class Environment(abc.ABC):
+    """The main API for garage environments.
+
+    The main public API methods are:
+        * step()
+        * reset()
+        * render()
+        * visualize()
+        * close()
+
+    Set the following attributes:
```

Comment:
please make these properties rather than attributes. that way they are immutable.